File: app/src/main/java/ru/aradxxx/ciceronetabssample/di/AppComponent.java
package ru.aradxxx.ciceronetabssample.di;

import android.app.Application;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.AndroidInjector;
import dagger.android.support.AndroidSupportInjectionModule;
import dagger.android.support.DaggerApplication;
import ru.aradxxx.ciceronetabssample.App;

@Singleton
@Component(modules = {
        AndroidSupportInjectionModule.class,
        ActivityBindingModule.class,
        NavigationModule.class,
})
public interface AppComponent extends AndroidInjector<DaggerApplication> {
    void inject(App app);

    @Component.Builder
    interface Builder {
        @BindsInstance
        Builder bindApplication(Application app);

        AppComponent build();
    }
}

File: app/src/main/java/com/github/aradxxx/ciceroneflowssample/di/AppComponent.kt
package com.github.aradxxx.ciceroneflowssample.di

import android.app.Application
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.DaggerApplication
import dagger.android.support.AndroidSupportInjectionModule
import com.github.aradxxx.ciceroneflowssample.App
import javax.inject.Singleton

@Singleton
@Component(modules = [
    AndroidSupportInjectionModule::class,
    ActivityBindingModule::class,
    NavigationModule::class
])
interface AppComponent : AndroidInjector<DaggerApplication> {
    fun inject(app: App)

    @Component.Builder
    interface Builder {
        @BindsInstance
        fun bindApplication(app: Application): Builder
        fun build(): AppComponent
    }
}
